system/mp/mpcomm.cpp

```diff
@@ -1639,6 +1639,13 @@ class CMPPacketReader: public ISocketSelectNotify, public CInterface
                     remaining = hdr.size-sizeof(hdr);
                     activemsg = new CMessageBuffer(remaining); // will get from low level IO at some stage
                     activeptr = (byte *)activemsg->reserveTruncate(remaining);
+
+                    // Ensure MP packet use peer endpoint
+                    SocketEndpoint peerEp;
+                    sock->getPeerEndpoint(peerEp);
+                    peerEp.port = hdr.sender.port;
```

Comment:
this and the similar line in the corresponding line in CMPConnectThread::run is preserving the original port,
but it shouldn't be here, I may have inadvertently reintroduced it.
At one point, subscriptions were failing because they relied on port matching the original (Client) endpoint.

I'll need to remove, retest